src/services/catalog.service.ts

```diff
@@ -0,0 +1,198 @@
+            breakLoops = true;
+            return false;
+          }
+        }
+      });
+    });
+    return retSubCategory;
+  }
+}
+
+interface IServiceItem {
+  iconClass: string;
+  imageUrl: string;
+  name: string;
+  catsBySubCats: any;
+  description: string;
+  longDescription: string;
+  resource: any;
+}
+
+export class ServiceItem implements IServiceItem {
+  public iconClass: string;
+  public imageUrl: string;
+  public name: string;
+  public catsBySubCats: any;
+  public description: string;
+  public longDescription: string;
+  public resource: any;
+  private catalogSrv: CatalogService;
+
+  constructor (serviceClass: any, catalogSrv: CatalogService) {
+    this.resource = serviceClass;
+    this.catalogSrv = catalogSrv;
+    this.imageUrl = this.getImage();
+    this.name = this.getName();
+    this.description = this.getDescription();
+    this.longDescription = this.getLongDescription();
+    this.catsBySubCats = this.getCategoriesBySubCategories();
+  }
+
+  private getImage() {
+    if (this.resource.metadata) {
+      return this.resource.metadata.imageUrl;
+    }
+  }
+
+  private getName() {
+    if (this.resource.osbMetadata) {
+      return this.resource.osbMetadata.displayName ? this.resource.osbMetadata.displayName : this.resource.metadata.name;
+    } else {
+      return this.resource.metadata.name;
+    }
+  }
+
+  private getDescription() {
+    if (this.resource.osbMetadata) {
+      return this.resource.osbMetadata.description;
+    }
+  }
+
+  private getLongDescription() {
+    if (this.resource.osbMetadata) {
+      return this.resource.osbMetadata.longDescription;
+    }
+  }
+
+  private getCategoriesBySubCategories() {
+    let catsBySubCats = {};
+
+    if (this.resource.osbTags) {
+      _.each(this.resource.osbTags, (subCategory) => {
+        let subCat = this.catalogSrv.getSubCategoryFromAlias(subCategory);
+        let cat = this.catalogSrv.getCategoryFromSubCategory(subCat);
+        catsBySubCats[subCat] = cat;
+      });
+    }
+    return catsBySubCats;
+  }
+}
+
+export class ImageItem implements IServiceItem {
+  public iconClass: string;
+  public imageUrl: string;
+  public name: string;
+  public catsBySubCats: any;
+  public description: string;
+  public longDescription: string;
+  public resource: any;
+  public isBuilder: boolean;
+  private catalogSrv: CatalogService;
+
+  constructor (image: any, catalogSrv : CatalogService) {
+    this.resource = image;
+    this.catalogSrv = catalogSrv;
+    this.iconClass = this.getIcon();
+    this.name = this.getName();
+    this.description = this.getDescription();
+    this.longDescription = this.getLongDescription();
+    this.catsBySubCats = this.getCategoriesBySubCategories();
+  }
+
+  private getIcon() {
+    return 'font-icon ' + this.catalogSrv.$filter('imageStreamTagIconClass')(this.resource);
+  }
+
+  private getName() {
+    return this.catalogSrv.$filter('annotation')(this.resource, "displayName");
```

Comment:
@dtaylor113 If you use the `displayName` filter instead of `annotation`, it will do that for you.

```js
return this.catalogSrv.$filter('displayName')(this.resource);
```